src/test/java/seedu/address/logic/commands/CreateTagCommandTest.java

```diff
@@ -0,0 +1,84 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+import static seedu.address.testutil.TypicalTagNames.VALID_TAGNAME_COLLEAGUES;
+import static seedu.address.testutil.TypicalTagNames.VALID_TAGNAME_NEIGHBOURS;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code CreateTagCommand}.
+ */
+public class CreateTagCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_validTagName_success() {
+        CreateTagCommand createTagCommand = new CreateTagCommand(VALID_TAGNAME_NEIGHBOURS);
+        String expectedMessage = String.format(CreateTagCommand.MESSAGE_CREATE_TAG_SUCCESS, VALID_TAGNAME_NEIGHBOURS);
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.addTag(new Tag(VALID_TAGNAME_NEIGHBOURS));
+        assertCommandSuccess(createTagCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_sameTagNameDifferentCase_throwsCommandException() {
+        Tag newTag = new Tag(VALID_TAGNAME_NEIGHBOURS);
+        model.addTag(newTag);
+        CreateTagCommand createTagCommand1 = new CreateTagCommand("frIenDS");
+        String expectedMessage = CreateTagCommand.MESSAGE_DUPLICATE_TAG;
+        assertCommandFailure(createTagCommand1, model, expectedMessage);
+    }
+
+    @Test
+    public void execute_tagDoesNotExistInModel_success() {
```

Comment:
I believe this test case is identical to `execute_validTagName_success`

Reply:
I agree with you that they are inherently 2 different tests that just happen to have the same execution! However, I feel we can make this to our advantage by combining them into a single test, effectively killing 2 birds with 1 stone. We could denote within the method using comments, or renaming the test method, to highlight that the test is meant to handle both test cases. 😀

Reply:
Yup that sounds like a great idea 👍